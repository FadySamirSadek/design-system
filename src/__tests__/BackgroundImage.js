import React from 'react'
import renderer from 'react-test-renderer'
import { BackgroundImage } from '..'

describe('BackgroundImage', () => {
  test('renders', () => {
    const json = renderer.create(<BackgroundImage />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with image', () => {
    const json = renderer.create(<BackgroundImage image="hello.png" />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-image', 'url(hello.png)')
  })

  test('renders with height', () => {
    const json = renderer.create(<BackgroundImage height="256px" />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('height', '256px')
  })
})
