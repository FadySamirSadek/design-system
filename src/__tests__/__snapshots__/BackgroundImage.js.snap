// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackgroundImage renders 1`] = `
.c0 {
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
  background-color: #dde1e4;
}

<div
  className="c0"
/>
`;

exports[`BackgroundImage renders with height 1`] = `
.c0 {
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
  background-color: #dde1e4;
  height: 256px;
}

<div
  className="c0"
  height="256px"
/>
`;

exports[`BackgroundImage renders with image 1`] = `
.c0 {
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
  background-color: #dde1e4;
  background-image: url(hello.png);
}

<div
  className="c0"
/>
`;
